Essentials/src/com/earth2me/essentials/register/payment/methods/ReserveEco.java

```diff
@@ -0,0 +1,289 @@
+package com.earth2me.essentials.register.payment.methods;
+
+
+import com.earth2me.essentials.register.payment.Method;
+import net.tnemc.core.Reserve;
+import net.tnemc.core.economy.EconomyAPI;
+import org.bukkit.plugin.Plugin;
+
+import java.math.BigDecimal;
+
+public class ReserveEco implements Method {
+
+    Plugin plugin;
+    EconomyAPI economy;
+
+    /**
+     * Encodes the Plugin into an Object disguised as the Plugin. If you want the original Plugin
+     * Class you must cast it to the correct Plugin, to do so you have to verify the name and or
+     * version then cast.
+     * <p/>
+     * <pre>
+     *    if (method.getName().equalsIgnoreCase("iConomy"))
+     *     iConomy plugin = ((iConomy)method.getPlugin());</pre>
+     *
+     * @return <code>Object</code>
+     *
+     * @see #getName()
+     * @see #getVersion()
+     */
+    @Override
+    public Plugin getPlugin() {
+        return plugin;
+    }
+
+    /**
+     * Returns the actual name of this method.
+     *
+     * @return <code>String</code> Plugin name.
+     */
+    @Override
+    public String getName() {
+        return "Reserve";
+    }
+
+    public String getProvider() {
+        return economy == null ? "No Provider" : economy.name();
+    }
+
+    /**
+     * Returns the reported name of this method.
+     *
+     * @return <code>String</code> Plugin name.
+     */
+    @Override
+    public String getLongName() {
+        return "Reserve Economy: " + getProvider();
+    }
+
+    /**
+     * Returns the actual version of this method.
+     *
+     * @return <code>String</code> Plugin version.
+     */
+    @Override
+    public String getVersion() {
+        return plugin.getDescription().getVersion();
+    }
+
+    /**
+     * Returns the amount of decimal places that get stored NOTE: it will return -1 if there is no
+     * rounding
+     *
+     * @return <code>int</code> for each decimal place
+     */
+    @Override
+    public int fractionalDigits() {
```

Comment:
This seems clowny but :man_shrugging:

Reply:
Does Essentials have rounding? This isn't a Reserve method so I'm not familiar with it, the vault class returns 0.

Reply:
As long as you only use BigDecimal methods, there shouldn't be any rounding.

Reply:
So then -1 is correct based off the javadoca here.